File: AI/event.cpp
#include "event.h"

Event::Event(const unsigned char threadIdx_, EventEnum type_):
	m_threadIdx(threadIdx_), m_type(type_)
{
}

unsigned char Event::threadIdx() const
{
	return m_threadIdx;
}

Event::EventEnum Event::type() const
{
	return m_type;
}

File: PyDarkLogic/AI/event.py
from enum import Enum

class Event(object):
    class EventEnum(Enum):
        START=0
        STOP=1
        STOP_THREAD=2
    def __init__(self, threadIdx, type):
        self._threadIdx=threadIdx
        self._type=type

    def threadIdx(self):
        return self._threadIdx

    def type(self):
        return self._type